### PR TITLE
Add TypeScript Declarations File

### PR DESCRIPTION
**Pull request by MatthewCallis**
- Add TypeScript declarations file and add to `package.json`.
 - Fix documentation comment for `canShowInteraction`.
 - ESLint fixes to platform specific files (missing `;`).

[SEA-690](https://apptentive.atlassian.net/browse/SEA-690)

### CODE REVIEW

File: js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// Type definitions for apptentive-react-native
+// Project: apptentive-react-native
+// Definitions by: Matthew Callis
+
+export = ApptentiveConfiguration;
```
Comment (dvonlehman) [inline js/index.d.ts]: Authoring TS declarations is still a bit of a black art, but reassigning `export` to `Apptentive` would then "hide" `ApptentiveConfiguration`? At least if it obeys the laws of JavaScript. There is probably multiple syntaxes that work, but I think you could have a single `export {ApptentiveConfiguration, Apptentive}`. You might need to try importing with a dummy `.ts` file to verify.

File: js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    static removeCustomPersonData(key: string): Promise<any>;
+    static addCustomDeviceData(key: string, value: any): Promise<any>;
+    static removeCustomDeviceData(key: string): Promise<any>;
+    static logIn(token: string): Promise<any>;
+    static logOut(): Promise<any>;
+    static onUnreadMessageCountChanged(value?: number): void;
+    static onAuthenticationFailed(value?: string): void;
+}
+
+declare class ApptentivePlatformSpecific {
```
Comment (weeeBox) [inline js/index.d.ts]: We probably should not export this

**Comment by weeeBox** [comment]
@MatthewCallis, each PR should go to `develop` (only the build system pushes to master through PR)

File: js/index.d.ts (hunk 2 above)
Comment (MatthewCallis) [reply js/index.d.ts]: There is nothing exported there.

Comment (weeeBox) [reply js/index.d.ts]: What I meant, we don't need  `ApptentivePlatformSpecific` in TypeScript

**Pull request merged**